src/NodeVisitor/LoopNodeVisitor.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\NodeVisitor;
+
+use Twig\Environment;
+use Twig\Error\SyntaxError;
+use Twig\Node\BlockNode;
+use Twig\Node\BreakNode;
+use Twig\Node\ForNode;
+use Twig\Node\MacroNode;
+use Twig\Node\Node;
+use Twig\Node\WithNode;
+
+final class LoopNodeVisitor implements NodeVisitorInterface
+{
+    /**
+     * @var int
+     */
+    private $depth = 0;
+
+    /**
+     * @var int[]
+     */
+    private $structureStack = [];
+
+    /**
+     * @var Node|null
+     */
+    private $elseNode;
+
+    public function enterNode(Node $node, Environment $env): Node
+    {
+        if ($this->isStructureNode($node)) {
+            $this->structureStack[] = [$node, $this->depth, $this->elseNode];
+            $this->depth = 0;
+        } elseif ($node instanceof ForNode) {
+            $this->elseNode = $node->hasNode('else') ? $node->getNode('else') : null;
+            ++$this->depth;
+        } elseif ($node instanceof BreakNode) {
+            $target = $node->getAttribute('target');
+
+            if (!\is_int($target) || $target < 1) {
+                throw new SyntaxError(sprintf('Break target must be an integer > 1, got "%s".', var_export($target, true)), $node->getTemplateLine(), $node->getSourceContext());
+            }
+
+            if (null !== $this->elseNode) {
+                if (1 === $target) {
+                    throw new SyntaxError('Cannot break from itself in "else" of "for" .', $node->getTemplateLine(), $node->getSourceContext());
+                }
+
+                $node->setTarget(--$target);
+            }
+
+            if ($target > $this->depth) {
+                if (\count($this->structureStack) > 0) {
```

Comment:
`if (0 < \count($this->structureStack)) {`

Reply:
:+1:  done, happy to follow any CS